### PR TITLE
Support template branches when creating apps

### PR DESCRIPTION
**Pull request by paulomarg**
### WHY are these changes introduced?

We'll have multiple options of the Remix app template going forward, and we want users to be able to select between a TS or JS version of the template.

### WHAT is this pull request doing?

Setting up the form so it asks a second question for the remix template, so the user can select a branch.

I implemented this as a generic way of adding branch options so that other templates can benefit from this in the future.

### How to test your changes?

- Run `dev create-app`, and select the remix template
- OR run `dev create-app --template remix`

image

### Post-release steps

Merge [this template PR](https://github.com/Shopify/shopify-app-template-remix/pull/372) which makes `main` the TS version of the code.

### Measuring impact

How do we know this change was effective? Please choose one:

- [x] Existing analytics will cater for this addition, since I'm just altering the template string we use in the end.

### Checklist

- [x] I've considered possible cross-platform impacts (Mac, Linux, Windows)
- [x] I've considered possible [documentation](https://shopify.dev) changes

### CODE REVIEW

File: packages/create-app/src/commands/init.ts [hunk 1]
```diff
@@ -83,7 +83,7 @@ export default class Init extends Command {
         'Only GitHub repository references are supported, ' +
           'e.g., https://github.com/Shopify/<repository>/[subpath]#[branch]',
       )
-    if (!url && !allTemplates.includes(template))
+    if (!url && !isPredefinedTemplate(template))
       throw new AbortError(
         outputContent`Only ${visibleTemplates
           .map((alias) => outputContent`${outputToken.yellow(alias)}`.value)
```
Comment (isaacroldan) [inline packages/create-app/src/commands/init.ts:89]: Should we also add a new flag to select the `branch`? but called flavor like for extensions
`create-app --template remix --flavor typescript`

Comment (paulomarg) [reply packages/create-app/src/commands/init.ts:89]: The way I set this up you'd pass in the template and get prompted for the branch, for `remix`. We could add a flag, but I (slightly) prefer the current experience.

Happy to go with whatever the team prefers.

Comment (isaacroldan) [reply packages/create-app/src/commands/init.ts:89]: There is a "rule" that all prompts should have an alternative without user interaction (for nonTTY terminals for instance), but that's already covered by the `--template` flag if you use a full repo name like `https://github.com/Shopify/<repository>/[subpath]#[branch]` 🤔 
So not sure if it's worth it to add a new flag or not, probably a @MeredithCastile decission

**Comment by paulomarg** [comment]
Re-requested review because I changed a few things!

**Comment by MeredithCastile** [comment]
Copying content recommendations over from our Slack thread:

Prompt:
> For your Remix template, which language do you want to use?
> > TypeScript
> > JavaScript

Error (using the Ink error pattern in the CLI UI Kit):
> Invalid option for --flavor.
> The --flavor flag for Remix accepts only `typescript` or `javascript` options.

**Pull request merged**